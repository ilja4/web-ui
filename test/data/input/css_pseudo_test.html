<!DOCTYPE html>
<!--
Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <link rel="import" href="component_css_pseudo.html">
  <script src="packages/web_ui/testing/testing.js"></script>
  <title>CSS Custom Pseudo-Element Test</title>
</head>
<body>
  <style>
    .test-1::x-background {
      background-color: red;
      color: yellow;
      font-weight: bold;
    }

    .test-2::x-background > * {
      background-color: blue;
      color: green;
      font-size: 36px !important;
    }
  </style>

  <x-comp-pseudo class=test-1></x-comp-pseudo>
  <x-comp-pseudo></x-comp-pseudo>
  <x-comp-pseudo class=test-2></x-comp-pseudo>

  <script type="application/dart">
import 'dart:async';
import 'dart:html';
import 'package:unittest/unittest.dart';
import 'package:web_ui/web_ui.dart';

main() {
  useShadowDom = false;

  Timer.run(() {
    // Set of components used.
    var urls = queryAll('link')
        .where((link) => link.attributes['rel'] == 'import')
        .map((link) => link.attributes['href'])
        .toSet();

    expect(urls.length, 1);
    expect(urls, contains('component_css_pseudo.html'));

    window.postMessage('done', '*');
   });
}
  </script>
</body>
</html>