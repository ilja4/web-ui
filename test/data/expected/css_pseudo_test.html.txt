Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x91
  RenderBlock {HTML} at (0,0) size 800x91
    RenderBody {BODY} at (8,8) size 784x75
      RenderBlock (anonymous) at (0,0) size 784x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,0) size 784x18
        RenderBlock {DIV} at (0,0) size 784x18 [color=#FFFF00] [bgcolor=#FF0000]
          RenderBlock {DIV} at (0,0) size 784x18
            RenderText {#text} at (0,0) size 40x18
              text run at (0,0) width 40: "Test"
      RenderBlock (anonymous) at (0,18) size 784x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,18) size 784x18
        RenderBlock {DIV} at (0,0) size 784x18 [bgcolor=#808080]
          RenderBlock {DIV} at (0,0) size 784x18
            RenderText {#text} at (0,0) size 40x18
              text run at (0,0) width 40: "Test"
      RenderBlock (anonymous) at (0,36) size 784x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,36) size 784x39
        RenderBlock {DIV} at (0,0) size 784x39 [bgcolor=#808080]
          RenderBlock {DIV} at (0,0) size 784x39 [color=#008000] [bgcolor=#0000FF]
            RenderText {#text} at (0,0) size 88x39
              text run at (0,0) width 88: "Test"
      RenderBlock (anonymous) at (0,75) size 784x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
